mobile/src/main/java/rs/readahead/washington/mobile/views/activity/FormSubmitActivity.kt

```diff
@@ -0,0 +1,352 @@
+        }
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu): Boolean {
+        menuInflater.inflate(R.menu.form_submit_menu, menu)
+        enableMenuItems(menu)
+        return true
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        val id = item.itemId
+        if (id == android.R.id.home) {
+            if (submitModel.isReSubmitting()) {
+                showStandardSheet(
+                    this.supportFragmentManager,
+                    getString(R.string.Collect_DialogTitle_StopExit),
+                    getString(R.string.Collect_DialogExpl_ExitingStopSubmission),
+                    getString(R.string.Collect_DialogAction_KeepSubmitting),
+                    getString(R.string.Collect_DialogAction_StopAndExit),
+                    null
+                ) { onDialogBackPressed() }
+
+                /*DialogsUtil.showExitWithSubmitDialog(this,
+                        (dialog, which) -> finish(),
+                        (dialog, which) -> {
+                        });*/
+            } else {
+                finish()
+            }
+            return true
+        }
+        return super.onOptionsItemSelected(item)
+    }
+
+    @Deprecated("Deprecated in Java")
+    override fun onBackPressed() {
+        if (submitModel.isReSubmitting()) {
+            showStandardSheet(
+                this.supportFragmentManager,
+                getString(R.string.Collect_DialogTitle_StopExit),
+                getString(R.string.Collect_DialogExpl_ExitingStopSubmission),
+                getString(R.string.Collect_DialogAction_StopAndExit),
+                getString(R.string.Collect_DialogAction_KeepSubmitting),
+                { onDialogBackPressed() },
+                null
+            )
+        } else {
+            super.onBackPressed()
+        }
+        finish()
+    }
+
+    private fun onDialogBackPressed() {
+        //MyApplication.bus().post(CollectFormSubmitStoppedEvent())
+        SharedLiveData.updateViewPagerPosition.postValue(OUTBOX_LIST_PAGE_INDEX)
+        super.onBackPressed()
+        return
+    }
+
+    override fun onPause() {
+        super.onPause()
+        if (submitModel.isReSubmitting()) {
+            submitModel.stopReSubmission()
+            submissionStoppedByUser()
+        }
+    }
+
+    private fun init() {
+        content.submitButton.setOnClickListener { view: View? ->
+            onSubmitClick()
+        }
+        content.cancelButton.setOnClickListener { view: View? ->
+            onCancelClick()
+        }
+        content.stopButton.setOnClickListener { view: View? -> onStopClick() }
+
+        viewModel.collectFormInstance.observe(this) { instance ->
+            if (instance != null) {
+                onGetFormInstanceSuccess(instance)
+            }
+        }
+
+        viewModel.onError.observe(this) { throwable ->
+            onGetFormInstanceError(throwable)
+        }
+
+        with(submitModel) {
```

Comment:
maybe it is better to use a function just to initObservers to maintain a clear code